### DISCUSSION TITLE
Aliased production build says "preview" activity

### DISCUSSION THREAD

**Discussion by santiher** [vercel/community#4364] [Feature Question]
### Question

I am using this flow:

- Build every feature branch in the **preview** environment.
- Build our production branch in the **production** environment.

In order to deploy staging I alias our staging domain to our preview build.
In order to deploy production I alias our production domain to our production build.

This flow was recommended to us previously by support in order to have 0 wait 0 downtime deploys.

The thing is that when I look into our logs it says that all the activity is coming from **preview**.
Even requests to our production site.

Is that correct? Is our "production" vercel deployment actually running in "preview"?
Is there a way to have it running in "production" that way?
Having to hit "deploy" at deploy time and wait for it to build is not very nice to me... Specially since I had to change our flow to follow the suggested practice.

**Comment by amyegan** [comment]
Hi @santiher. Anything deployed to the `preview` environment should run in `preview` and anything deployed to the `production` environment should run in `production`.

If you're able to share links to some example deployments here, I would like to take a closer look to help figure out why your **production** domain is reporting `preview` environment activity.

**Reply by santiher** [reply]
Hi Amy,

This is our production build: https://vercel.com/everydotorg/website/GJyDmfrniLbAUkUb9CiUsm7Kbv9H
And our preview build: https://vercel.com/everydotorg/website/CVKzUm4jYr993dHXLLdZ3yhoB74m

Of the same commit.

Another thing I'm noticing now is that the `environment` field of the deployment is grayed out as if loading in the production build:

<img width="256" alt="Screenshot 2023-10-05 at 14 22 10" src="https://github.com/vercel/community/assets/4341390/2a23c541-2642-4177-9841-a299e2ea5f96">

It does say preview in the preview one:

<img width="247" alt="Screenshot 2023-10-05 at 14 22 49" src="https://github.com/vercel/community/assets/4341390/d899cb45-8759-48bb-b333-822e7c61014c">

And just in case it changes something we don't have our domains added to the project's settings because we alias it manually at deploy time instead of it being aliased to the latest build automatically.

**Reply by amyegan** [reply]
You mentioned "having to hit 'deploy' at deploy time and wait for it to build", so it sounds like you're using a [custom workflow](https://vercel.com/guides/using-vercel-cli-for-custom-workflows) with a CI/CD provider and [Vercel CLI](https://vercel.com/docs/cli). If that's the case, do you have a _preview_ workflow using `vercel build` and separate _production_ workflow using `vercel build --prod`? Or do you have it configured differently?

**Reply by santiher** [reply]
Nope, vercel is linked to github and builds production when we push to the production branch and build preview when pushing to any other branch.

Hitting the deploy button was more figuratively speaking, I think it happens automatically if you link the domains directly in the project settings.

**Reply by amyegan** [reply]
Thanks for clarifying that! I'm reaching out to the logs engineering team to get their advice on what might be causing this
